components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90

```diff
@@ -773,47 +584,271 @@ subroutine ma_convproc_sh_intr(                 &
+
+
+!=========================================================================================
+subroutine load_updraft_massflux(               &
+                icol,   cmfmcsh,                & ! in
+                tot_conv_layer,   jt, maxg, mu  ) ! out
+!-----------------------------------------------------------------------------
+! load updraft mass flux from cmfmcsh
+!-----------------------------------------------------------------------------
+
+   real(r8), intent(in)    :: cmfmcsh(pcols,pverp) ! Shallow conv mass flux [kg/m2/s]
+   integer,  intent(in)    :: icol             ! index of column
+   integer,  intent(out)   :: tot_conv_layer   ! total layers of convection in this column
+   integer,  intent(out)   :: jt(pcols)        ! Index of cloud top for each column
+   integer,  intent(out)   :: maxg(pcols)      ! Index of cloud bot for each column
+   real(r8), intent(out)   :: mu(pcols,pver)   ! Updraft mass flux (positive) [mb/s]
+
+
+   integer              :: kk
+   real(r8), parameter  :: small_massflux = 1.0e-7_r8
+   ! if mass-flux < 1e-7 kg/m2/s ~= 1e-7 m/s ~= 1 cm/day, treat as zero
+
+   ! initiate variables
+   tot_conv_layer = 0 ! total layers of convection in this column
+   mu(:,:) = 0.0_r8
+   jt(:) = -1
+   maxg(:) = -1
+
+   do kk = 2, pver
+       if (cmfmcsh(icol,kk) >= small_massflux) then
+            ! mu has units of mb/s
+            mu(icol,kk) = cmfmcsh(icol,kk) / hund_ovr_g
+            tot_conv_layer = tot_conv_layer + 1
+            if (tot_conv_layer == 1) jt(icol) = kk - 1
+            maxg(icol) = kk
+       endif
+   enddo ! kk
+
+end subroutine load_updraft_massflux
+
+!=========================================================================================
+subroutine extend_belowcloud_downward(             &
+                        icol,     dpdry,           & ! in
+                        maxg,     mu,              & ! inout
+                        maxg_init                  ) ! out
+!-----------------------------------------------------------------------------
+! extend below-cloud source region downwards
+!-----------------------------------------------------------------------------
+   integer,  intent(in)    :: icol              ! index of column
+   real(r8), intent(in)    :: dpdry(pcols,pver) ! layer delta-p-dry [mb]
+   integer,  intent(inout) :: maxg(pcols)       ! Index of cloud bot for each column
+   real(r8), intent(inout) :: mu(pcols,pver)    ! Updraft mass flux (positive) [mb/s]
+   integer,  intent(out)   :: maxg_init         ! initial maxg at icol
+
+   integer              :: kk, kaa, kbb         ! vertical index
+   real(r8)             :: dp_sum               ! sum of dpdry for weighting purpose
+   integer              :: maxg_minval
+
+   ! initiate variables
+   maxg_minval = pver*2  ! this variable seems not used
+   maxg_minval = min( maxg_minval, maxg(icol) )
+
+   kaa = maxg(icol)          ! cloud bot level
+   kbb = min( kaa+4, pver )  ! 4 levels below cloud bottom
```

Comment:
Can we change the variable names to `cld_bot_lev` and `cld_bot_below_4_levs` (or something better) instead of `kaa` and `kbb`?

Reply:
While we're talking about variable names, what about `jtop` instead of `jt` and `jbot` instead of `maxg`?

Reply:
maxg and jt are firstly introduced in zm_conv scheme and used in many other subroutines. I agree that their physical meaning isn't quite clear, but hesitate to change them as it will break the consistency. @singhbalwinder do you have any comment on it? 
I'll change variables names of kaa and kbb.